Ical.Net/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -954,8 +954,10 @@ public override HashSet<Period> Evaluate(IDateTime referenceDate, DateTime perio
             // This case is not defined by RFC 5545. We handle it by evaluating the rule
             // as if referenceDate had a time (i.e. set to midnight).
 
-            referenceDate = referenceDate.Copy<IDateTime>();
-            referenceDate.HasTime = true;
+            if (referenceDate.HasTime)
```

Comment:
This is a redundant check, because already checked in the previous line.

Reply:
correct